File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int left = 0, right = nums.size() - 1;

        while(left <= right){
            // cout << left << " " << right << endl;
            if(nums[left] <= nums[right]) return nums[left];
            int mid = left + (right - left) / 2;
            if(nums[mid] < nums[right]){
                right = mid;
            }
            else{
                left = mid + 1;
            }
            
        }
        return -1;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        mid = max(piles)
        left = 0
        right = mid
        res = 0
        while True:
            print(mid)
            total_time = sum([math.ceil(x / mid) for x in piles])
            if total_time <= h:
                res = mid
                right = mid
            else:
                left = mid
            
            mid = left + (right - left) // 2
            if left >= right - 1:
                break

        return res
            